src/bidiMapper/utils/log.ts

```diff
@@ -15,16 +15,18 @@
  */
 export function log(type: string): (...message: any[]) => void {
   return (...messages: any[]) => {
-    const elementId = type + '_log';
+    // If run in browser, add debug message to the page.
+    if (typeof window !== 'undefined' && window?.document?.documentElement) {
+      console.log(type, ...messages);
 
-    if (!window.document.getElementById(elementId)) {
-      window.document.documentElement.innerHTML += `<h3>${type}:</h3><pre id='${elementId}'></pre>`;
-      window.document.getElementById(elementId);
-    }
+      const elementId = type + '_log';
 
-    const element = window.document.getElementById(elementId);
+      if (!window.document.getElementById(elementId)) {
+        window.document.documentElement.innerHTML += `<h3>${type}:</h3><pre id='${elementId}'></pre>`;
```

Comment:
It looks like this doesn't support more than one `type`, because all the other content will get removed by this line if there's a new `type`.

If multiple `type`s should be supported, maybe rework this to create and add a new section to the document?

Reply:
Oh wait, I misunderstood the original code, `+=` here is key.